lib/url_parser.js

```diff
@@ -4,9 +4,121 @@ var ReadPreference = require('./read_preference'),
   parser = require('url'),
   f = require('util').format,
   assign = require('./utils').assign,
-  Logger = require('mongodb-core').Logger;
+  Logger = require('mongodb-core').Logger,
+  dns = require('dns');
 
-module.exports = function(url, options) {
+module.exports = function(url, options, callback) {
+  if (typeof options === 'function') (callback = options), (options = {});
+  options = options || {};
+
+  var result = parser.parse(url, true);
+  if (result.protocol !== 'mongodb:' && result.protocol !== 'mongodb+srv:') {
+    return callback(new Error('invalid schema, expected mongodb or mongodb+srv'));
```

Comment:
nitpick: let's decide on whether we should use capitalized letters for error messages or not, the newly added messages in this section do both.

Reply:
Yes! This was on my mind to discuss later too, I was generally trying to do what was already there which seemed to be mostly lowercase but was definitely not consistent.